src/Microsoft.Diagnostics.Runtime/src/Common/ParallelObjectSet.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.Diagnostics.Runtime
 {
-    internal class ParallelObjectSet : ObjectSet
+    public class ParallelObjectSet : ObjectSet
```

Comment:
Why is this public?  It's an implementation detail of the GCRoot code.  It was not meant for general consumption...  I don't expect external people to rebuild the GCRoot code from scratch, and if they did I would imagine they'd use a different abstraction.

Also, it's not supported to call ClrMD in any multithreaded environment (other than very narrow cases, like the GCRoot code), it would be a bit confusing to mark public something that I didn't intend others to use and only has a very narrow acceptable usage.

Reply:
I ended up backing out use of this class so personally don't need it to be public anymore. I've reverted this to be internal again.